### PR TITLE
Add "See this course in <first|second> semester" button to course overview in sidebar

### LINKED ISSUE #85: For a first term course, add the option to switch to the second term version of it, and vice versa.
For instance, if I selected a course, say MAT223H1F from search, and there is a second semester version available, MAT223H1S, then I should be able to see a link style button above Remove that says "See course in second semester". Clicking on this button takes me to the second semester offering of the course. The same applies to a course in second semester, such as MAT223H1S. 

However, if a course:
- is a full year course, or 
- does not have an alternate offering in this year,
I should not see this button appear.

### PR DESCRIPTION
**Pull request by kaspar-p**
Fixes #85 

- Added a checkCourseExists query
- Didn't do any error handling for that query, I don't know how...
- Added t() strings for button text. All in English except for German.
- Move `[chosenCourse, setChosenCourse]` up to the Sidebar from SearchView, since CourseView needs it.
  - Isn't it redundant though? Why can the `onClick` handlers not just use the context dispatch, with action type `SET_SIDEBAR_COURSE`. Why use in-component state at all?
  - Maybe it used to be useful as a purely SearchView thing but now that it got lifted it's not? Or maybe it is? I don't know.
- Can I please get a review on the aesthetics @eamonma

### CODE REVIEW

File: components/sidebar/SearchView/SearchView.tsx [hunk 1]
```diff
@@ -34,20 +34,25 @@ import SearchViewHints from "./SearchViewHints"
...
   setSearchQuery,
   searchOffset,
   setSearchOffset,
-}: {
-  searchQuery: string
-  setSearchQuery: Dispatch<React.SetStateAction<string>>
-  searchOffset: number
-  setSearchOffset: Dispatch<React.SetStateAction<number>>
-}) => {
+  chosenCourse,
```
Comment (eamonma) [inline components/sidebar/SearchView/SearchView.tsx:51]: If I remember correctly, the sidebar course is in the app context? Can you give a tldr of this new structure?

File: src/utils/misc.ts [hunk 2]
```diff
@@ -1 +1,3 @@
+export type Optional<T> = T | undefined // I'm pretty sure this exists somewhere and can't find it
```
Comment (eamonma) [inline src/utils/misc.ts:1]: Where do we need this

Comment (eamonma) [reply src/utils/misc.ts:1]: If this is used in an object type can we not just use 
```ts
key?: type
```

**Review by eamonma — CHANGES_REQUESTED** [review]

**Comment by eamonma** [comment]
> Can I please get a review on the aesthetics @eamonma
How does the new button look with the `link` variant instead of the `button` variant?

File: components/sidebar/CourseView.tsx [hunk 3]
```diff
@@ -61,9 +69,47 @@ const CourseView = ({ setSearchQuery }: { setSearchQuery: Function }) => {
...
+      if (siblingCourseId === null) return
+
+      const result = await checkCourseExists({
+        variables: {
+          id: siblingCourseId,
+        },
+      })
+
+      if (result.error) {
+        // I don't know how to show errors in sqrl :/ so return for now
```
Comment (eamonma) [inline components/sidebar/CourseView.tsx:104]: I think this failing is not the end of the world for now.

File: src/utils/misc.ts (hunk 2 above)
Comment (kaspar-p) [reply src/utils/misc.ts:1]: You're right, I never used it. I was meaning to use it in the `siblingCourseId` type. I'll get rid of the definition. Watch this video though it's poggers:

[https://www.youtube.com/watch?v=C2w45qRc3aU](https://www.youtube.com/watch?v=C2w45qRc3aU)

File: components/sidebar/SearchView/SearchView.tsx [hunk 4]
```diff
@@ -34,20 +34,25 @@ import SearchViewHints from "./SearchViewHints"
 const MotionFlex = motion<FlexProps>(Flex)
 const MotionButton = motion<ButtonProps>(Button)
 
+interface SearchViewProps {
```
[the remark this answers is not in the document]
Comment (kaspar-p) [reply components/sidebar/SearchView/SearchView.tsx:37]: Ok, I'll change this.

File: components/sidebar/CourseView.tsx [hunk 5]
```diff
@@ -61,9 +69,47 @@ const CourseView = ({ setSearchQuery }: { setSearchQuery: Function }) => {
...
+        return
```
[the remark this answers is not in the document]
Comment (kaspar-p) [reply components/sidebar/CourseView.tsx:105]: Ok.

File: components/sidebar/CourseView.tsx (hunk 3 above)
Comment (kaspar-p) [reply components/sidebar/CourseView.tsx]: Ok.

**Comment by eamonma** [comment]
> > > Can I please get a review on the aesthetics @eamonma
> 
> > How does the new button look with the `link` variant instead of the `button` variant? @kaspar-p
> 
> @eamonma Bad, I think. It just looks like regular text and then underlines black on hover. I think the button looks better:
> 
> `Link` component from chakra: Screen Shot 2022-07-13 at 1 36 40 PM
> 
> `Button` component from chakra: Screen Shot 2022-07-13 at 1 37 24 PM
> 
> I prefer the Button component. If that's not what you meant by link variant, let me know.

Still use the Button component, but set a prop `variant="link"`

**Comment by kaspar-p** [comment]
Meh, still don't like it. What do you think:
Screen Shot 2022-07-13 at 2 01 20 PM

It has a hover underline the same color as the text

**Review by eamonma — APPROVED** [review]
Looks good.

**Pull request merged**